numpy/random/tests/test_smoke.py

```diff
@@ -92,403 +93,469 @@ def warmup(rg, n=None):
         assert_(len(vals) == 10)
 
     def test_noncentral_f(self):
-        vals = self.rg.noncentral_f(3, 1000, 2, 10)
+        data = self._create_gen()
+        vals = data.rg.noncentral_f(3, 1000, 2, 10)
         assert_(len(vals) == 10)
-        vals = self.rg.noncentral_f(np.array([3] * 10), 1000, 2)
+        vals = data.rg.noncentral_f(np.array([3] * 10), 1000, 2)
         assert_(len(vals) == 10)
-        vals = self.rg.noncentral_f(3, np.array([1000] * 10), 2)
+        vals = data.rg.noncentral_f(3, np.array([1000] * 10), 2)
         assert_(len(vals) == 10)
-        vals = self.rg.noncentral_f(3, 1000, np.array([2] * 10))
+        vals = data.rg.noncentral_f(3, 1000, np.array([2] * 10))
         assert_(len(vals) == 10)
 
     def test_normal(self):
-        vals = self.rg.normal(10, 0.2, 10)
+        data = self._create_gen()
+        vals = data.rg.normal(10, 0.2, 10)
         assert_(len(vals) == 10)
 
     def test_pareto(self):
-        vals = self.rg.pareto(3.0, 10)
+        data = self._create_gen()
+        vals = data.rg.pareto(3.0, 10)
         assert_(len(vals) == 10)
 
     def test_poisson(self):
-        vals = self.rg.poisson(10, 10)
+        data = self._create_gen()
+        vals = data.rg.poisson(10, 10)
         assert_(len(vals) == 10)
-        vals = self.rg.poisson(np.array([10] * 10))
+        vals = data.rg.poisson(np.array([10] * 10))
         assert_(len(vals) == 10)
-        params_1(self.rg.poisson)
+        params_1(data.rg.poisson)
 
     def test_power(self):
-        vals = self.rg.power(0.2, 10)
+        data = self._create_gen()
+        vals = data.rg.power(0.2, 10)
         assert_(len(vals) == 10)
 
     def test_integers(self):
-        vals = self.rg.integers(10, 20, 10)
+        data = self._create_gen()
+        vals = data.rg.integers(10, 20, 10)
         assert_(len(vals) == 10)
 
     def test_rayleigh(self):
-        vals = self.rg.rayleigh(0.2, 10)
+        data = self._create_gen()
+        vals = data.rg.rayleigh(0.2, 10)
         assert_(len(vals) == 10)
-        params_1(self.rg.rayleigh, bounded=True)
+        params_1(data.rg.rayleigh, bounded=True)
 
     def test_vonmises(self):
-        vals = self.rg.vonmises(10, 0.2, 10)
+        data = self._create_gen()
+        vals = data.rg.vonmises(10, 0.2, 10)
         assert_(len(vals) == 10)
 
     def test_wald(self):
-        vals = self.rg.wald(1.0, 1.0, 10)
+        data = self._create_gen()
+        vals = data.rg.wald(1.0, 1.0, 10)
         assert_(len(vals) == 10)
 
     def test_weibull(self):
-        vals = self.rg.weibull(1.0, 10)
+        data = self._create_gen()
+        vals = data.rg.weibull(1.0, 10)
         assert_(len(vals) == 10)
 
     def test_zipf(self):
-        vals = self.rg.zipf(10, 10)
+        data = self._create_gen()
+        vec_1d = np.arange(2.0, 102.0)
+        vec_2d = np.arange(2.0, 102.0)[None, :]
+        mat = np.arange(2.0, 102.0, 0.01).reshape((100, 100))
```

Comment:
this is the inlined content of `_extra_setup` and this matches what was in that function